account/watcher/controller.go

```diff
@@ -47,9 +47,8 @@ type Config struct {
 	HandleAccountExpiry func(*btcec.PublicKey, uint32) error
 }
 
-// Watcher is responsible for the on-chain interaction of an account, whether
-// that is confirmation or spend.
-type Watcher struct {
+// controller implements the Controller interface
+type controller struct {
```

Comment:
Add compile time assertion that we satisfy the controller interface?
`var _ Controller = (*controller)(nil)`